movie_library/components/movie_component.py

```diff
@@ -0,0 +1,63 @@
+from ..repositories.movie_repository import MovieRepository
+from ..models.movie import Movie
+from rest_framework import status
+from rest_framework.response import Response
+from django.core.exceptions import BadRequest
+from django.shortcuts import render, get_object_or_404
+
+
+class MovieComponent():
+    """
+    this used for applying any needed logic
+    """
+
+    @staticmethod
+    def get_movies():
+        # no login needed
+        return MovieRepository().get_movies()
+
+    @staticmethod
+    def get_movie(pk):
+        queryset = MovieComponent.get_movies()
+        movie = get_object_or_404(queryset, pk=pk)
```

Comment:
good idea to use shortcuts 
also you can add this logic in ur component and repository